src/AbstractDescriptor.php

```
@@ -46,7 +46,12 @@
         /** @psalm-suppress MixedAssignment */
         $this->def = $descriptor->def ?? $descriptor->ref ?? $descriptor->src ?? null;
         /** @psalm-suppress MixedAssignment */
-        $this->doc = $descriptor->doc ?? null;
+        // Old:
+-        $this->doc = $descriptor->doc->value ?? null;
++        // Safely handle stdClass|string|null for $descriptor->doc
++        $this->doc = (is_object($descriptor->doc) && property_exists($descriptor->doc, 'value'))
++            ? $descriptor->doc->value
```

PHPStan (2.0.3)

Syntax error, unexpected '?' on line 53


(phpstan.parse)